ibis/expr/analysis.py

```diff
@@ -463,6 +407,17 @@ def _base_table(table_node):
         return _base_table(table_node.table.op())
 
 
+def _predicate_schema_matches_table_schema(predicate, schema):
```

Comment:
I'm not sure if this is where this should be done. As I'm typing this I'm thinking that it's not. This will limit further nested substitutions. Instead, we can check the schema in `_subs` and do nothing if a sub isn't possible. That way, more deeply nested expressions can take advantage of the pushdown.